spec/engine/miq_ae_method_service/miq_ae_service_methods_spec.rb

```diff
@@ -163,6 +163,53 @@ def category_create_script
     expect(ct.entries.collect(&:name).include?('fred')).to be_truthy
   end
 
+  context "#tag_delete" do
+    before(:each) do
+      @ct = FactoryBot.create(:classification_department_with_tags)
+      @ct_name = @ct.name
+      @entry_name = @ct.entries.first.name
+      @assignment_tag = "/chargeback_rate/assigned_to/vm/tag/managed/#{@ct_name}/#{@entry_name}"
+      Tag.create!(:name => @assignment_tag)
+    end
```

Comment:
The `before(:each)` is an older syntax and should just be `before` now.

I think the whole section would could be done a little differently:

```ruby
context "#tag_delete" do
  let(:ct) { FactoryBot.create(:classification_department_with_tags) }
  let(:entry_name) { ct.entries.first.name }
  let(:assignment_tag) { "/chargeback_rate/assigned_to/vm/tag/managed/#{ct.name}/#{entry_name}" }
  before { Tag.create!(:name => assignment_tag) }

  ...
```

I removed the `ct_name` variable as I don't think it is really needed.  Changing to `let` blocks also means the variable references in the tests no longer require the `@` reference.

Reply:
Got it! let looks much better